### ISSUE TITLE
feature/manage diagnostic devices (CRUD)

### ISSUE THREAD

**Issue by chris-dare** [serenity-health/roadmap#104]
**Tell us about your request!**
Lab machines are a form of devices used at healthcare centers to analyze laboratory test specimen collected from patients. They are extremely helpful because they speed up the investigation process while improving the accuracy of test results

**Which app does this concern?**
Provider portal

**What will be benefit of this feature?**
Providers can register devices used in processing samples. 
Test results can be received from those devices to facilitate the delivery of diagnostic investigations

**Describe the solution you'd like**
1. Complete the UI designs to add and view diagnostic devices (@agbavorrichmond)
2. Implement the FHIR device specification: https://www.hl7.org/fhir/device.html (@TuyizereBapt)
a. Create the API spec (under `diagnostics.views` and `provider.diagnostics.urls`) and implement this
3. Implement the screens to view devices, add and update a device (@pkdadson)
4. Integrate the device management API from @TuyizereBapt - @pkdadson 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by pkdadson** [comment]
@agbavorrichmond please any update of the UI designs?

**Comment by agbavorrichmond** [comment]
Yes, there is buts its not been confirmed yet m, will update you once it has been confirmed

**Comment by pkdadson** [comment]
Alright @agbavorrichmond , I'm on standby.

**Comment by pkdadson** [comment]
@agbavorrichmond how far?

**Comment by chris-dare** [comment]
@agbavorrichmond share the loom and the link to the screens with @pkdadson

**Comment by agbavorrichmond** [comment]
@pkdadson 
1. here is the link to the video demo of the  lab machine integrations https://www.loom.com/share/e8df70cc865b45b9bfae5bb347775f97?sharedAppSource=personal_library and the link to the figma file is https://www.figma.com/file/f29U89yv9q1HvYVaStLIl9/SERENITY-EMR-(PRODUCTION-%E2%9C%85)?node-id=4067%3A35265
2. The print prescription design is the first thing on this page https://www.figma.com/file/f29U89yv9q1HvYVaStLIl9/SERENITY-EMR-(PRODUCTION-%E2%9C%85)?node-id=1594%3A19469

**Comment by TuyizereBapt** [comment]
API endpoint for creating a Device: https://serenity-health.readme.io/reference/diagnostics-workflow#create-a-device

**Comment by dexios1** [comment]
> API endpoint for creating a Device: https://serenity-health.readme.io/reference/diagnostics-workflow#create-a-device

Hi Dadson, the API for diagnostic device crud is ready.

@TuyizereBapt I just realized the PR in https://github.com/serenity-health/core-systems-api/pull/50 closed this automatically.

**Comment by chris-dare** [comment]
> API endpoint for creating a Device: https://serenity-health.readme.io/reference/diagnostics-workflow#create-a-device

Hi @pkdadson, the API for diagnostic device crud is ready.

@TuyizereBapt I just realized the PR in https://github.com/serenity-health/core-systems-api/pull/50 closed this automatically.

**Comment by chris-dare** [comment]
> @pkdadson
> 
> 1. here is the link to the video demo of the  lab machine integrations https://www.loom.com/share/e8df70cc865b45b9bfae5bb347775f97?sharedAppSource=personal_library and the link to the figma file is https://www.figma.com/file/f29U89yv9q1HvYVaStLIl9/SERENITY-EMR-(PRODUCTION-%E2%9C%85)?node-id=4067%3A35265
> 2. The print prescription design is the first thing on this page https://www.figma.com/file/f29U89yv9q1HvYVaStLIl9/SERENITY-EMR-(PRODUCTION-%E2%9C%85)?node-id=1594%3A19469

Hi @TuyizereBapt,

Dadson needs the API to conform to the UI we're are implementing. If there are necessary changes i.e. additional fields as well, then these need to be added to the UI based on the FHIR spec. We need:
1. What the required fields should be
2. Provision of valuesets like [device type](https://www.hl7.org/fhir/valueset-device-type.html)
3. Status reason valueset
4. Status valueset

**Comment by pkdadson** [comment]
@TuyizereBapt please we need a value set for device names to help create a device. 
@dexios1 can fill you in on this.

**Comment by pkdadson** [comment]
@dexios1 please can we split this card so we pay attention to the usecase.
